pkg/oauth/apiserver/registry/useroauthaccesstokens/delegate/proxywatcher.go

```diff
@@ -0,0 +1,112 @@
+package delegate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+
+	oauthapi "github.com/openshift/oauth-apiserver/pkg/oauth/apis/oauth"
+)
+
+// OAuthAccessTokenWatcher implements a Watcher interface to watch OAuthAccessTokens
+// and filter them to match only those for the current user
+type OAuthAccessTokenWatcher struct {
```

Comment:
is this taken from anywhere?

Reply:
I was looking for inspiration in the watchers of k8s.io/apimachinery/pkg/watch but I ended up mixing some of the basic bits of the openshift-apiserver's project watcher with custom resource proxying logic